openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/FairGBMDescriptorUtil.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2022 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.feedzai.openml.provider.lightgbm;
+
+import com.feedzai.openml.provider.descriptor.ModelParameter;
+import com.feedzai.openml.provider.descriptor.fieldtype.ChoiceFieldType;
+import com.feedzai.openml.provider.descriptor.fieldtype.FreeTextFieldType;
+import com.feedzai.openml.provider.descriptor.fieldtype.NumericFieldType;
+import com.google.common.collect.ImmutableSet;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+
+/**
+ * Utility to organize all the necessary Machine Learning Hyper-Parameters for configuring the training of LightGBM.
+ *
+ * @author Andre Cruz (andre.cruz@feedzai.com)
+ * @since 1.2.1 // TODO: check what's the current version
+ */
+public class FairGBMDescriptorUtil extends LightGBMDescriptorUtil {
+
+    public static final String CONSTRAINT_GROUP_COLUMN_PARAMETER_NAME = "constraint_group_column";
+
+    /**
+     * Defines the set of model parameters accepted by the FairGBM algorithm.
+     */
+    static final Set<ModelParameter> PARAMS = Sets.union(ImmutableSet.of(
+            // The single parameter that will change for every different dataset
+            new ModelParameter(
+                    CONSTRAINT_GROUP_COLUMN_PARAMETER_NAME,
+                    "(Fairness) Sensitive group column",
+                    "Fairness constraints will be enforced over this column.\n"
+                            + "This column must be in categorical format.\n"
+                            + "Start this string with `name:` to use the name of a column here, \n"
+                            + "e.g., `name:age_group` for a column named `age_group`.",
+                    MANDATORY,
+                    new FreeTextFieldType("")
+//                    new FreeTextFieldType("", ".+")       # TODO: https://github.com/feedzai/feedzai-openml/issues/68
+            ),
+
+            new ModelParameter(
+                    "constraint_type",
+                    "(Fairness) Constraint type",
+                    "Enforces group-wise parity on the given target metric for the selected group column. "
+                            + "In general, FPR can be used for most fraud detection settings, "
```

Comment:
Fraud is just a part of what is considered financial crime, so I suggest removing it, so that we can broaden the scope for future application in more use cases. Does it make sense?

Reply:
yes I think it does